### PR TITLE
[State Sync] Small cleanups.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This PR offers three small cleanups to state sync (each in their own commit):
1. Clean up some of the state sync v2 logs (e.g., to avoid log spam under failure conditions).
2. Increase the frequency of yields for CPU/IO intensive threads that yield only infrequently (e.g., see: https://github.com/aptos-labs/aptos-core/issues/623)
3. Add a configurable value to set the max number of consecutive data notifications to process before yielding back to the main state sync driver loop. This should further help the above.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Existing tests cover this functionality.

## Related PRs

None, but this PR relates to: https://github.com/aptos-labs/aptos-core/issues/245

### CODE REVIEW

**Review by zekun000 — COMMENTED** [review]
hmm, why backpressure doesn't work? ideally if the execution goes too far, we should just wait(yield) until commit catches up?

**Comment by JoshLind** [comment]
Also, @zekun000, I've added another commit that helps get at the heart of the issue for the driver. We need the driver to be able to process commit and error notifications it receives. But, if the data is streaming continuously without a break, it can result in long periods of time before the main driver loop is engaged. After this PR, we should be able to just use `spawn_blocking` on the executor and committer and be good to go. But, I'll delay this fix until after the cut 😄

**Review by zekun000 — APPROVED** [review]
it sounds like we need more threads for state sync runtime? yield_now doesn't sounds like proper long term solution, happy to go over the overall flow and brainstorm better ideas. unblock for now

**Comment by JoshLind** [comment]
Sweet, thanks @zekun000!

> it sounds like we need more threads for state sync runtime? yield_now doesn't sounds like proper long term solution, happy to go over the overall flow and brainstorm better ideas. unblock for now

For sure, we'll sync offline 😄 

/land

**Pull request merged**
